src/datatypes.ts

```diff
@@ -153,6 +156,83 @@ export function deserializeDatetime(literalString: string): Date | null {
   return date;
 }
 
+/**
+ * @internal
+ * @param value Value to serialise.
+ * @returns String representation of `value`.
+ * @see https://www.w3.org/TR/xmlschema-2/#date-lexical-representation
+ */
+export function serializeDate(value: Date): string {
+
+  const year = value.getFullYear();
+  const month = value.getMonth();
+  const day = value.getDate();
+  const [_, timezone] = splitTimeFromTimezone(value.toISOString());
+  
+
+  return year + "-" + (month+1) + "-" + day + timezone;
+}
+
+/**
+ * @internal
+ * @param value Value to deserialise.
+ * @returns Deserialized datetime, or null if the given value is not a valid serialised datetime.
+ * @see https://www.w3.org/TR/xmlschema-2/#date-lexical-representation
+ */
+export function deserializeDate(literalString: string): Date | null {
+  // Date in the format described at
+  // https://www.w3.org/TR/xmlschema-2/#date-lexical-representation
+  // (without constraints on the value).
+  // -? - An optional leading `-`.
+  // \d{4,}- - Four or more digits followed by a `-` representing the year. Example: "3000-".
+  // \d\d-\d\d - Two digits representing the month and two representing the day of the month,
+  //             separated by a `-`. Example: "11-03".
+  // (Z|(\+|-)\d\d:\d\d) - Optionally, the letter Z indicating UTC, or a `+` or `-` followed by two digits for
+  //                       the hour offset and two for the minute offset, separated by a `:`.
+  //                       Example: "+13:37".
+
+  const dateRegEx =
+    /-?\d{4,}-\d\d-\d\d(Z|(\+|-)\d\d:\d\d)?/;
+  if (!dateRegEx.test(literalString)) {
+    return null;
+  }
+
+  const signedDateString = literalString;
+  // The date string can optionally be prefixed with `-`,
+  // in which case the year is negative:
+  const [yearMultiplier, dateString] =
+    signedDateString.charAt(0) === "-"
+      ? [-1, signedDateString.substring(1)]
+      : [1, signedDateString];
+  const [yearString, monthString, dayAndTimezoneString] = dateString.split("-");
+
+  const dayString = (dayAndTimezoneString.length > 2 ? dayAndTimezoneString.substring(0, 2) : dayAndTimezoneString);
+
+  const utcFullYear = Number.parseInt(yearString, 10) * yearMultiplier;
+  const utcMonth = Number.parseInt(monthString, 10) - 1;
+  const utcDate = Number.parseInt(dayString, 10);
+  const hour = 12;
```

Comment:
Ah right, we're basically stripping the timezone information from the input string, because there's no way to represent that using `Date`. Reeeaallly looking forward to JS getting a better date&time API :angry: